File: ZooJava/src/main/java/com/example/zoojava/models/Employee.java
package com.example.zoojava.models;

import javafx.beans.property.*;

import java.time.LocalDate;

/**
 * Clase de creación de los empleados del zoo.
 */
public class Employee {
    private StringProperty name;
    private StringProperty email;
    private StringProperty password;
    private ObjectProperty<LocalDate> birthDate;
    private BooleanProperty isAdmin;

    public Employee(String name, String email, String password, LocalDate birthDate, boolean isAdmin) {
        this.name = new SimpleStringProperty(name);
        this.email = new SimpleStringProperty(email);
        this.password = new SimpleStringProperty(password);
        this.birthDate = new SimpleObjectProperty<>(birthDate);
        this.isAdmin = new SimpleBooleanProperty(isAdmin);
    }


    public String getName() {
        return name.get();
    }

    public StringProperty nameProperty() {
        return name;
    }

    public void setName(String name) {
        this.name.set(name);
    }

    public String getEmail() {
        return email.get();
    }

    public StringProperty emailProperty() {
        return email;
    }

    public void setEmail(String email) {
        this.email.set(email);
    }

    public String getPassword() {
        return password.get();
    }

    public StringProperty passwordProperty() {
        return password;
    }

    public void setPassword(String password) {
        this.password.set(password);
    }

    public LocalDate getBirthDate() {
        return birthDate.get();
    }

    public ObjectProperty<LocalDate> birthDateProperty() {
        return birthDate;
    }

    public void setBirthDate(LocalDate birthDate) {
        this.birthDate.set(birthDate);
    }

    public boolean isIsAdmin() {
        return isAdmin.get();
    }

    public BooleanProperty isAdminProperty() {
        return isAdmin;
    }

    public void setIsAdmin(boolean isAdmin) {
        this.isAdmin.set(isAdmin);
    }

    public String toCsv() {
        return  getName()+";"+getEmail()+";"+getPassword()+";"+getBirthDate().toString()+";"+isIsAdmin();
    }
}

File: ZooKotlin/src/main/kotlin/com/example/zookotlin/models/Employee.kt
package com.example.zookotlin.models

import javafx.beans.property.*
import java.time.LocalDate

/**
 * Clase creacion modelo empleados.
 */
class Employee {
    private var name: StringProperty
    private var email: StringProperty
    private var password:StringProperty
    private var birthDate: ObjectProperty<LocalDate>
    private var isAdmin: BooleanProperty

    constructor(name: String, email:String, password:String, birthDate: LocalDate, isAdmin: Boolean){
        this.name = SimpleStringProperty(name)
        this.email = SimpleStringProperty(email)
        this.password = SimpleStringProperty(password)
        this.birthDate= SimpleObjectProperty(birthDate)
        this.isAdmin = SimpleBooleanProperty(isAdmin)
    }


    fun getName(): String {
        return name.get()
    }

    fun nameProperty(): StringProperty{
        return name
    }

    fun setName(name: String) {
        this.name.set(name)
    }

    fun getEmail(): String {
        return email.get()
    }

    fun emailProperty(): StringProperty {
        return email
    }

    fun setEmail(email: String) {
        this.email.set(email)
    }

    fun getPassword(): String{
        return password.get()
    }

    fun passwordProperty(): StringProperty {
        return password
    }

    fun setPassword(password: String) {
        this.password.set(password)
    }

    fun getBirthDate(): LocalDate {
        return birthDate.get()
    }

    fun birthDateProperty(): ObjectProperty<LocalDate> {
        return birthDate
    }

    fun setBirthDate(birthDate: LocalDate?) {
        this.birthDate.set(birthDate)
    }

    fun isIsAdmin(): Boolean {
        return isAdmin.get()
    }

    fun isAdminProperty(): BooleanProperty {
        return isAdmin
    }

    fun setIsAdmin(isAdmin: Boolean) {
        this.isAdmin.set(isAdmin)
    }

    fun toCsv():String{
        return getName() + ";" + getEmail() + ";" + getPassword() + ";" + getBirthDate().toString() + ";" + isIsAdmin()
    }
}